src/plugins/tbin/tbinplugin.cpp

```diff
@@ -0,0 +1,353 @@
+                }
+                layer->setCell(ix, iy, cell);
+
+                if (ttile.props.size() > 0)
+                {
+                    Tiled::MapObject* obj = new Tiled::MapObject("TileData", "", QPointF(ix * tlayer.tileSize.x, iy * tlayer.tileSize.y), QSizeF(1 * tlayer.tileSize.x, 1 * tlayer.tileSize.y));
+                    tbinToTiledProperties(ttile.props, obj);
+                    objects->addObject(obj);
+                }
+            }
+            map->addLayer(layer);
+            map->addLayer(objects);
+        }
+    }
+    catch (std::exception& e)
+    {
+        mError = tr((std::string("Exception: ") + e.what()).c_str());
+    }
+
+    return map;
+}
+
+bool TbinMapFormat::write(const Tiled::Map *map, const QString &fileName)
+{
+    try
+    {
+        tbin::Map tmap;
+        //tmap.id = map->name();
+        tiledToTbinProperties(map, tmap.props);
+
+        for(const Tiled::SharedTileset& tilesheet : map->tilesets())
+        {
+            tbin::TileSheet ttilesheet;
+            ttilesheet.id = tilesheet->name().toStdString();
+            ttilesheet.image = tilesheet->imageSource().toStdString();
+            ttilesheet.margin.x = ttilesheet.margin.y = tilesheet->margin();
+            ttilesheet.spacing.x = ttilesheet.spacing.y = tilesheet->tileSpacing();
+            ttilesheet.sheetSize.x = tilesheet->gridSize().width();
+            ttilesheet.sheetSize.y = tilesheet->gridSize().height();
+            ttilesheet.tileSize.x = tilesheet->tileSize().width();
+            ttilesheet.tileSize.y = tilesheet->tileSize().height();
+            tiledToTbinProperties(map, tmap.props);
+
+            // TODO: Per tile properties, investigate syntax
+
+            tmap.tilesheets.push_back(std::move(ttilesheet));
+        }
+
+        std::vector< Tiled::ObjectGroup* > objGroups;
+        std::map< std::string, tbin::Layer* > tileLayerIdMap;
+        for (Tiled::Layer* rawLayer : map->layers())
+        {
+            if (Tiled::ObjectGroup* layer = rawLayer->asObjectGroup())
+            {
+                objGroups.push_back(layer);
+            }
+            else if (Tiled::TileLayer* layer = rawLayer->asTileLayer())
+            {
+                tbin::Layer tlayer;
+                tlayer.id = layer->name().toStdString();
+                tlayer.layerSize.x = layer->size().width();
+                tlayer.layerSize.y = layer->size().height();
+                tlayer.tileSize.x = map->tileSize().width();
+                tlayer.tileSize.y = map->tileSize().height();
+                //tlayer.visible = ???;
+                for (int iy = 0; iy < tlayer.layerSize.y; ++iy)
+                {
+                    for (int ix = 0; ix < tlayer.layerSize.x; ++ix)
+                    {
+                        Tiled::Cell cell = layer->cellAt(ix, iy);
+                        tbin::Tile ttile;
+                        ttile.staticData.tileIndex = -1;
+
+                        if (cell.tile() != NULL)
```

Comment:
Please do this to avoid looking up the tile many times:
```c++
if (Tile *tile = cell.tile()) {
```

Coding style: please put the `{` after `if`, `for`, `catch`, etc. on the same line.